frontend/src/app/chart/chart.component.ts

```diff
@@ -0,0 +1,255 @@
+import { Component, OnInit, ViewEncapsulation } from '@angular/core';
+import { GetDataService } from '../services/get-data.service';
+import * as d3 from 'd3';
+
+@Component({
+  selector: 'main-chart',
+  templateUrl: './chart.component.html',
+  styleUrls: ['./chart.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class ChartComponent implements OnInit {
+  records: any;
```

Comment:
Please define records type. This is the variable that will hold the loaded data right? Please be careful about the type of variables that you will do calculations later. For example, I noticed you are doing value comparisons below, if you don't have data type defined correctly, it might cause unexpected results. For example, d3.min(['5','10']) = '5' if you don't use '+' operator into the comparing method.

Reply:
fixed